src/scp/Slot.cpp

```diff
@@ -107,32 +107,31 @@ Slot::getLatestMessage(NodeID const& id) const
 }
 
 bool
-Slot::isLatestStatement(SCPStatement const& st)
+Slot::shouldReplaceSt(SCPStatement const& st, SCPStatement const& hint)
 {
-    bool isLatest = true;
+    bool replace = false;
 
-    auto node = st.nodeID;
-    if (st.pledges.type() == SCPStatementType::SCP_ST_NOMINATE)
+    bool stNomination = st.pledges.type() == SCPStatementType::SCP_ST_NOMINATE;
+    bool hintNomination =
+        hint.pledges.type() == SCPStatementType::SCP_ST_NOMINATE;
+
```

Comment:
you can simplify the conditions below by just checking first that `stNomination == hintNomination`